src/SplStack/LegacyImplementation.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\Stdlib\SplStack;
+
+use Serializable;
+use SplStack;
+use UnexpectedValueException;
+
+use function get_class;
+use function gettype;
+use function is_array;
+use function is_object;
+use function serialize;
+use function sprintf;
+use function unserialize;
+
+/**
+ * Serializable version of SplStack
+ */
+class LegacyImplementation extends SplStack implements Serializable
```

Comment:
Should be `@internal` IMO: the implementation should not be part of the public API